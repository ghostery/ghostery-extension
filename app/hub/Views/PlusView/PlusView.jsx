/**
 * Plus View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import globals from '../../../../src/classes/Globals';

/**
 * @class Implement the Plus View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class PlusView extends Component {
	/**
	 * Helper render function for rendering the Plus Button
	 * @param  {Boolean} addSideMargin  whether to add spacing around the button
	 * @return {JSX} JSX of the Plus Button
	 */
	_renderButton = (additionalClasses) => {
		const { isPlus, onPlusClick } = this.props;
		const buttonHref = `${globals.CHECKOUT_BASE_URL}/plus?utm_source=gbe&utm_campaign=intro_hub_plus`;
		const buttonClassNames = ClassNames('PlusView__button', 'button', additionalClasses, {
			disabled: isPlus,
		});

		return isPlus ? (
			<div className={buttonClassNames}>
				{t('already_subscribed')}
			</div>
		) : (
			<a className={buttonClassNames} href={buttonHref} onClick={onPlusClick} target="_blank" rel="noopener noreferrer">
				{t('get_ghostery_plus')}
			</a>
		);
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Plus View of the Hub app
	 */
	render() {
		return (
			<div className="PlusView">
				<div className="row align-center">
					<div className="columns">
						<div className="PlusView__headingImage" />
						<div className="PlusView__headingTitle text-center">
							{t('hub_supporter_header_title')}
						</div>
						<div className="PlusView__headingDescription text-center">
							{t('hub_supporter_header_description')}
						</div>
						<div className="PlusView__costContainer flex-container align-middle align-justify">
							{this._renderButton()}
						</div>
					</div>
				</div>
				<div className="PlusView--addPaddingTop row small-up-1 large-up-3 align-center">
					<div className="PlusView__perk columns text-center">
						<div className="PlusView__perkIcon themes" />
						<div className="PlusView__perkTitle">
							{t('new_themes')}
						</div>
						<div className="PlusView__perkDescription">
							{t('hub_supporter_perk_themes_description')}
						</div>
					</div>
					<div className="PlusView__perk columns text-center">
						<div className="PlusView__perkIcon stats" />
						<div className="PlusView__perkTitle">
							{t('historical_blocking_stats')}
						</div>
						<div className="PlusView__perkDescription">
							{t('hub_supporter_perk_stats_description')}
						</div>
					</div>
					<div className="PlusView__perk columns text-center">
						<div className="PlusView__perkIcon support" />
						<div className="PlusView__perkTitle">
							{t('hub_supporter_perk_more_title')}
						</div>
						<div className="PlusView__perkDescription">
							{t('hub_supporter_perk_more_description')}
						</div>
					</div>
				</div>
				<div className="PlusView__manifestoContainer">
					<div className="PlusView__manifestoBackground row align-center">
						<div dangerouslySetInnerHTML={{ __html: t('hub_supporter_manifesto') }} className="PlusView__manifestoText columns small-12 medium-10 large-8 text-center" />
					</div>
				</div>
				<div className="PlusView--addPaddingTop PlusView--addPaddingBottom">
					<div className="row align-center">
						<div className="PlusView__feature small-12 medium-6 large-5 large-offset-1 columns">
							<div className="PlusView__headingTitle PlusView--addPaddingTop">
								{t('new_themes')}
							</div>
							<div className="PlusView__headingDescription">
								{t('hub_supporter_feature_theme_description')}
							</div>
							{this._renderButton('show-for-large')}
						</div>
						<div className="PlusView__feature small-12 medium-6 columns">
							<img
								className="PlusView__featureImage theme"
								src="/app/images/hub/plus/feature-theme.png"
								alt={t('new_themes')}
							/>
						</div>
					</div>
					<div className="PlusView--addPaddingTop hide-for-large">
						<div className="row align-center-middle">
							<div className="PlusView__bar flex-child-grow" />
							{this._renderButton('PlusView--addSideMargin')}
							<div className="PlusView__bar flex-child-grow" />
						</div>
					</div>
				</div>
				<div className="PlusView--addPaddingTop PlusView--addPaddingBottom PlusView--rowDarken">
					<div className="row align-center">
						<div className="PlusView__feature small-12 medium-6 large-5 columns show-for-large">
							<img
								className="PlusView__featureImage stats"
								src="/app/images/hub/plus/feature-stats.svg"
								alt={t('historical_blocking_stats')}
							/>
						</div>
						<div className="PlusView__feature small-12 medium-6 large-5 columns">
							<div className="PlusView__headingTitle">
								{t('historical_blocking_stats')}
							</div>
							<div className="PlusView__headingDescription">
								{t('hub_supporter_feature_stats_description')}
							</div>
							{this._renderButton('show-for-large')}
						</div>
						<div className="PlusView__feature columns hide-for-large">
							<img
								className="PlusView__featureImage stats"
								src="/app/images/hub/plus/feature-stats.svg"
								alt={t('historical_blocking_stats')}
							/>
						</div>
					</div>
					<div className="PlusView--addPaddingTop hide-for-large">
						<div className="row align-center-middle">
							<div className="PlusView__bar flex-child-grow" />
							{this._renderButton('PlusView--addSideMargin')}
							<div className="PlusView__bar flex-child-grow" />
						</div>
					</div>
				</div>
				<div className="PlusView--addPaddingTop PlusView--addPaddingBottom">
					<div className="row align-center">
						<div className="PlusView__feature small-12 medium-6 large-5 large-offset-1 columns">
							<div className="PlusView__headingTitle">
								{t('priority_support')}
							</div>
							<div className="PlusView__headingDescription">
								{t('hub_supporter_feature_support_description')}
							</div>
							{this._renderButton('show-for-large')}
						</div>
						<div className="PlusView__feature columns small-12 medium-6">
							<img
								className="PlusView__featureImage support"
								src="/app/images/hub/plus/feature-support.svg"
								alt={t('priority_support')}
							/>
						</div>
					</div>
					<div className="PlusView--addPaddingTop hide-for-large">
						<div className="row align-center-middle">
							<div className="PlusView__bar flex-child-grow" />
							{this._renderButton('PlusView--addSideMargin')}
							<div className="PlusView__bar flex-child-grow" />
						</div>
					</div>
				</div>
				<div className="PlusView--addPaddingTop PlusView--addPaddingBottom PlusView--rowDarken">
					<div className="row align-center">
						<div className="PlusView__feature columns medium-8 large-5 text-center">
							<div className="PlusView__headingTitle">
								{t('hub_supporter_feature_more_title')}
							</div>
							<div className="PlusView__headingDescription">
								{t('hub_supporter_feature_more_description')}
							</div>
							<img
								className="PlusView__featureImage more"
								src="/app/images/hub/plus/feature-more.svg"
								alt={t('hub_supporter_feature_more_title')}
							/>
						</div>
					</div>
					<div className="PlusView--addPaddingTop">
						<div className="row align-center-middle">
							<div className="PlusView__bar flex-child-grow" />
							{this._renderButton('PlusView--addSideMargin')}
							<div className="PlusView__bar flex-child-grow" />
						</div>
					</div>
				</div>
			</div>
		);
	}
}

// PropTypes ensure we pass required props of the correct type
PlusView.propTypes = {
	isPlus: PropTypes.bool.isRequired,
	onPlusClick: PropTypes.func.isRequired,
};

export default PlusView;
