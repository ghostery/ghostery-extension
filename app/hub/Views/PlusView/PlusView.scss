/**
* Plus View Sass
*
* Ghostery Browser Extension
* https://www.ghostery.com/
*
* Copyright 2019 Ghostery, Inc. All rights reserved.
*
* This Source Code Form is subject to the terms of the Mozilla Public
* License, v. 2.0. If a copy of the MPL was not distributed with this
* file, You can obtain one at http://mozilla.org/MPL/2.0
*/

// Plus View
.PlusView {
	margin: 40px 0 0;
	overflow-x: hidden;
}
.PlusView--addPaddingTop {
	padding-top: 40px;
}
.PlusView--addPaddingBottom {
	padding-bottom: 40px;
}
.PlusView--addSideMargin {
	margin-left: 35px;
	margin-right: 35px;
}
.PlusView--rowDarken {
	background-color: #f7f7f7;
}
.PlusView__costContainer {
	margin: 0 auto;
	max-width: 182px;
}
.PlusView__button {
	min-width: 180px;
	margin-bottom: 0;
	line-height: 20.2px;
	border-top: none;
	border-bottom: none;
	border-right-color: #9ecc42;
	border-left-color: #429321;
	background: linear-gradient(to right, #429321,  #9ecc42);
}
.PlusView__button.disabled,
.PlusView__button.disabled:hover {
	background: #154700;
	border: 1px solid #154700;
	line-height: 18.2px;
}
.PlusView__headingImage {
	height: 195px;
	margin-left: -55px;
	margin-bottom: 30px;
	background-image: url('/app/images/hub/plus/plus-logo.svg');
	background-repeat: no-repeat;
	background-position: center center;
}
.PlusView__headingTitle {
	font-size: 24px;
	font-weight: 700;
	line-height: 38px;
	color: #4a4a4a;
}
.PlusView__headingDescription {
	font-size: 14px;
	line-height: 38px;
	color: #4a4a4a;
	margin-bottom: 20px;
}
.PlusView__headingCost {
	font-size: 14px;
	color: #9b9b9b;
}
.PlusView__headingCost sup {
	padding: 0 2px;
	font-size: 30px;
	font-weight: 500;
	color: #4a4a4a;
}
.PlusView__headingCost span {
	padding-right: 11px;
	font-size: 64px;
	font-weight: 500;
	color: #4a4a4a;
}
.PlusView__perk {
	padding: 30px;
}
.PlusView__perkTitle {
	font-size: 14px;
	font-weight: 700;
	line-height: 38px;
	margin-bottom: 15px;
	color: #4a4a4a;
}
.PlusView__perkDescription {
	font-size: 14px;
	line-height: 30px;
	color: #4a4a4a;
}
.PlusView__perkIcon {
	height: 80px;
	width: 80px;
	margin: 0 auto 15px;
	background-position: center center;
	background-repeat: no-repeat;
}
.PlusView__perkIcon.themes {
	background-image: buildPerkThemeIcon(#a8ca59, #0a0a0a);
}
.PlusView__perkIcon.stats {
	background-image: buildPerkStatsIcon(#a8ca59, #0a0a0a);
}
.PlusView__perkIcon.support {
	background-image: buildPerkSupportIcon(#a8ca59, #0a0a0a);
}
.PlusView__featureImage.theme {
	max-width: 420px;
	margin-top: -20px;
}
.PlusView__featureImage.support{
	max-width: 275px;
}
.PlusView__manifestoContainer {
	margin: 80px 0 0;
	padding-top: 40px;
	background: linear-gradient(to bottom left, #ecd868, #d4af37);
}
.PlusView__manifestoBackground {
	padding-bottom: 20px;
	background-image: url('/app/images/hub/plus/manifesto-background.svg');
	background-size: 100%;
}
.PlusView__manifestoText {
	font-size: 24px;
	font-weight: 700;
	line-height: 40px;
	color: #ffffff;
	padding: 50px 0;
}
.PlusView__bar {
	height: 1px;
	background-color: #9b9b9b;
}
@media only screen and (max-width: 960px) {
	.PlusView__perkIcon {
		margin-bottom: 10px;
	}
	.PlusView__perkTitle {
		margin-bottom: 10px;
	}
	.PlusView__manifestoContainer {
		margin-top: 60px;
	}
	.PlusView__manifestoBackground {
		padding-bottom: 10px;
	}
	.PlusView__manifestoText {
		margin-top: -20px;
		font-size: 20px;
		line-height: 30px;
		padding: 50px 60px;
	}
	.PlusView__featureImage.theme {
		margin: 30px -20px 15px 0;
		max-width: 350px;
	}
	.PlusView__featureImage.support {
		margin: 0 0 -10px -33px;
		max-width: 240px;
	}
	.PlusView__feature {
		text-align: center;
	}
	.PlusView__headingDescription {
		line-height: 30px;
		margin-top: 4px;
	}
	.PlusView__costContainer {
		margin-top: 24px;
	}
}
@media only screen and (max-width: 500px) {
	.PlusView__bar {
		display: none;
	}
}
