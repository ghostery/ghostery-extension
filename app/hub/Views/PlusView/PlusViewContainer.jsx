/**
 * Plus View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import PlusView from './PlusView';

/**
 * @class Implement the Plus View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class PlusViewContainer extends Component {
	constructor(props) {
		super(props);

		const title = t('hub_supporter_page_title');
		window.document.title = title;

		props.actions.getUser();
	}

	/**
	 * Sends the necessary ping to background
	 */
	_sendPlusPing = () => {
		this.props.actions.sendPing({ type: 'plus_cta_hub' });
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Plus View of the Hub app
	 */
	render() {
		const { user } = this.props;
		const childProps = {
			isPlus: user && user.plusAccess || false,
			onPlusClick: this._sendPlusPing,
		};

		return <PlusView {...childProps} />;
	}
}

// PropTypes ensure we pass required props of the correct type
PlusViewContainer.propTypes = {
	user: PropTypes.shape({
		email: PropTypes.string,
		plusAccess: PropTypes.bool,
	}),
	actions: PropTypes.shape({
		sendPing: PropTypes.func.isRequired,
		getUser: PropTypes.func.isRequired,
	}).isRequired,
};

// Default props used in the Plus View
PlusViewContainer.defaultProps = {
	user: {
		email: false,
		plusAccess: false,
	},
};

export default PlusViewContainer;
