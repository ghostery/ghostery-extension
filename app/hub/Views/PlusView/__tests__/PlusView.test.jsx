/**
 * Plus View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import PlusView from '../PlusView';

describe('app/hub/Views/PlusView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('plus view is rendered correctly when the user is not a plus', () => {
			const initialState = {
				isPlus: false,
				onPlusClick: () => {},
			};

			const component = renderer.create(
				<PlusView {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('plus view is rendered correctly when the user is a plus', () => {
			const initialState = {
				isPlus: true,
				onPlusClick: () => {},
			};

			const component = renderer.create(
				<PlusView {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				isSignedIn: false,
				isPlus: false,
				onPlusClick: jest.fn(),
			};

			const component = shallow(<PlusView {...initialState} />);
			expect(component.find('.PlusView').length).toBe(1);
			expect(component.find('.PlusView__headingImage').length).toBe(1);
			expect(component.find('.PlusView__headingTitle').length).toBe(5);
			expect(component.find('.PlusView__headingDescription').length).toBe(5);
			expect(component.find('.PlusView__perk').length).toBe(3);
			expect(component.find('.PlusView__perkIcon').length).toBe(3);
			expect(component.find('.PlusView__perkTitle').length).toBe(3);
			expect(component.find('.PlusView__perkDescription').length).toBe(3);
			expect(component.find('.PlusView__manifestoContainer').length).toBe(1);
			expect(component.find('.PlusView__manifestoBackground').length).toBe(1);
			expect(component.find('.PlusView__manifestoText').length).toBe(1);
			expect(component.find('.PlusView__featureImage').length).toBe(5);

			expect(initialState.onPlusClick.mock.calls.length).toBe(0);
			component.find('.PlusView__button').first().simulate('click');
			expect(initialState.onPlusClick.mock.calls.length).toBe(1);

			expect(component.find('.PlusView__button').first().props().href).toBe('https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus')

			expect(component.find('.PlusView__button').length).toBe(8);
			expect(component.find('.PlusView__button.disabled').length).toBe(0);
			component.setProps({ isPlus: true });
			expect(component.find('.PlusView__button').length).toBe(8);
			expect(component.find('.PlusView__button.disabled').length).toBe(8);
		});
	});
});
