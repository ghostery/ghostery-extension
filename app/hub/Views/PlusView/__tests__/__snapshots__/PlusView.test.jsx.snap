// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/PlusView component Snapshot tests with react-test-renderer plus view is rendered correctly when the user is a plus 1`] = `
<div
  className="PlusView"
>
  <div
    className="row align-center"
  >
    <div
      className="columns"
    >
      <div
        className="PlusView__headingImage"
      />
      <div
        className="PlusView__headingTitle text-center"
      >
        hub_supporter_header_title
      </div>
      <div
        className="PlusView__headingDescription text-center"
      >
        hub_supporter_header_description
      </div>
      <div
        className="PlusView__costContainer flex-container align-middle align-justify"
      >
        <div
          className="PlusView__button button disabled"
        >
          already_subscribed
        </div>
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop row small-up-1 large-up-3 align-center"
  >
    <div
      className="PlusView__perk columns text-center"
    >
      <div
        className="PlusView__perkIcon themes"
      />
      <div
        className="PlusView__perkTitle"
      >
        new_themes
      </div>
      <div
        className="PlusView__perkDescription"
      >
        hub_supporter_perk_themes_description
      </div>
    </div>
    <div
      className="PlusView__perk columns text-center"
    >
      <div
        className="PlusView__perkIcon stats"
      />
      <div
        className="PlusView__perkTitle"
      >
        historical_blocking_stats
      </div>
      <div
        className="PlusView__perkDescription"
      >
        hub_supporter_perk_stats_description
      </div>
    </div>
    <div
      className="PlusView__perk columns text-center"
    >
      <div
        className="PlusView__perkIcon support"
      />
      <div
        className="PlusView__perkTitle"
      >
        hub_supporter_perk_more_title
      </div>
      <div
        className="PlusView__perkDescription"
      >
        hub_supporter_perk_more_description
      </div>
    </div>
  </div>
  <div
    className="PlusView__manifestoContainer"
  >
    <div
      className="PlusView__manifestoBackground row align-center"
    >
      <div
        className="PlusView__manifestoText columns small-12 medium-10 large-8 text-center"
        dangerouslySetInnerHTML={
          Object {
            "__html": "hub_supporter_manifesto",
          }
        }
      />
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature small-12 medium-6 large-5 large-offset-1 columns"
      >
        <div
          className="PlusView__headingTitle PlusView--addPaddingTop"
        >
          new_themes
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_theme_description
        </div>
        <div
          className="PlusView__button button show-for-large disabled"
        >
          already_subscribed
        </div>
      </div>
      <div
        className="PlusView__feature small-12 medium-6 columns"
      >
        <img
          alt="new_themes"
          className="PlusView__featureImage theme"
          src="/app/images/hub/plus/feature-theme.png"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop hide-for-large"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <div
          className="PlusView__button button PlusView--addSideMargin disabled"
        >
          already_subscribed
        </div>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom PlusView--rowDarken"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature small-12 medium-6 large-5 columns show-for-large"
      >
        <img
          alt="historical_blocking_stats"
          className="PlusView__featureImage stats"
          src="/app/images/hub/plus/feature-stats.svg"
        />
      </div>
      <div
        className="PlusView__feature small-12 medium-6 large-5 columns"
      >
        <div
          className="PlusView__headingTitle"
        >
          historical_blocking_stats
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_stats_description
        </div>
        <div
          className="PlusView__button button show-for-large disabled"
        >
          already_subscribed
        </div>
      </div>
      <div
        className="PlusView__feature columns hide-for-large"
      >
        <img
          alt="historical_blocking_stats"
          className="PlusView__featureImage stats"
          src="/app/images/hub/plus/feature-stats.svg"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop hide-for-large"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <div
          className="PlusView__button button PlusView--addSideMargin disabled"
        >
          already_subscribed
        </div>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature small-12 medium-6 large-5 large-offset-1 columns"
      >
        <div
          className="PlusView__headingTitle"
        >
          priority_support
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_support_description
        </div>
        <div
          className="PlusView__button button show-for-large disabled"
        >
          already_subscribed
        </div>
      </div>
      <div
        className="PlusView__feature columns small-12 medium-6"
      >
        <img
          alt="priority_support"
          className="PlusView__featureImage support"
          src="/app/images/hub/plus/feature-support.svg"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop hide-for-large"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <div
          className="PlusView__button button PlusView--addSideMargin disabled"
        >
          already_subscribed
        </div>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom PlusView--rowDarken"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature columns medium-8 large-5 text-center"
      >
        <div
          className="PlusView__headingTitle"
        >
          hub_supporter_feature_more_title
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_more_description
        </div>
        <img
          alt="hub_supporter_feature_more_title"
          className="PlusView__featureImage more"
          src="/app/images/hub/plus/feature-more.svg"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <div
          className="PlusView__button button PlusView--addSideMargin disabled"
        >
          already_subscribed
        </div>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`app/hub/Views/PlusView component Snapshot tests with react-test-renderer plus view is rendered correctly when the user is not a plus 1`] = `
<div
  className="PlusView"
>
  <div
    className="row align-center"
  >
    <div
      className="columns"
    >
      <div
        className="PlusView__headingImage"
      />
      <div
        className="PlusView__headingTitle text-center"
      >
        hub_supporter_header_title
      </div>
      <div
        className="PlusView__headingDescription text-center"
      >
        hub_supporter_header_description
      </div>
      <div
        className="PlusView__costContainer flex-container align-middle align-justify"
      >
        <a
          className="PlusView__button button"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop row small-up-1 large-up-3 align-center"
  >
    <div
      className="PlusView__perk columns text-center"
    >
      <div
        className="PlusView__perkIcon themes"
      />
      <div
        className="PlusView__perkTitle"
      >
        new_themes
      </div>
      <div
        className="PlusView__perkDescription"
      >
        hub_supporter_perk_themes_description
      </div>
    </div>
    <div
      className="PlusView__perk columns text-center"
    >
      <div
        className="PlusView__perkIcon stats"
      />
      <div
        className="PlusView__perkTitle"
      >
        historical_blocking_stats
      </div>
      <div
        className="PlusView__perkDescription"
      >
        hub_supporter_perk_stats_description
      </div>
    </div>
    <div
      className="PlusView__perk columns text-center"
    >
      <div
        className="PlusView__perkIcon support"
      />
      <div
        className="PlusView__perkTitle"
      >
        hub_supporter_perk_more_title
      </div>
      <div
        className="PlusView__perkDescription"
      >
        hub_supporter_perk_more_description
      </div>
    </div>
  </div>
  <div
    className="PlusView__manifestoContainer"
  >
    <div
      className="PlusView__manifestoBackground row align-center"
    >
      <div
        className="PlusView__manifestoText columns small-12 medium-10 large-8 text-center"
        dangerouslySetInnerHTML={
          Object {
            "__html": "hub_supporter_manifesto",
          }
        }
      />
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature small-12 medium-6 large-5 large-offset-1 columns"
      >
        <div
          className="PlusView__headingTitle PlusView--addPaddingTop"
        >
          new_themes
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_theme_description
        </div>
        <a
          className="PlusView__button button show-for-large"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
      </div>
      <div
        className="PlusView__feature small-12 medium-6 columns"
      >
        <img
          alt="new_themes"
          className="PlusView__featureImage theme"
          src="/app/images/hub/plus/feature-theme.png"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop hide-for-large"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <a
          className="PlusView__button button PlusView--addSideMargin"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom PlusView--rowDarken"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature small-12 medium-6 large-5 columns show-for-large"
      >
        <img
          alt="historical_blocking_stats"
          className="PlusView__featureImage stats"
          src="/app/images/hub/plus/feature-stats.svg"
        />
      </div>
      <div
        className="PlusView__feature small-12 medium-6 large-5 columns"
      >
        <div
          className="PlusView__headingTitle"
        >
          historical_blocking_stats
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_stats_description
        </div>
        <a
          className="PlusView__button button show-for-large"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
      </div>
      <div
        className="PlusView__feature columns hide-for-large"
      >
        <img
          alt="historical_blocking_stats"
          className="PlusView__featureImage stats"
          src="/app/images/hub/plus/feature-stats.svg"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop hide-for-large"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <a
          className="PlusView__button button PlusView--addSideMargin"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature small-12 medium-6 large-5 large-offset-1 columns"
      >
        <div
          className="PlusView__headingTitle"
        >
          priority_support
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_support_description
        </div>
        <a
          className="PlusView__button button show-for-large"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
      </div>
      <div
        className="PlusView__feature columns small-12 medium-6"
      >
        <img
          alt="priority_support"
          className="PlusView__featureImage support"
          src="/app/images/hub/plus/feature-support.svg"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop hide-for-large"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <a
          className="PlusView__button button PlusView--addSideMargin"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
  <div
    className="PlusView--addPaddingTop PlusView--addPaddingBottom PlusView--rowDarken"
  >
    <div
      className="row align-center"
    >
      <div
        className="PlusView__feature columns medium-8 large-5 text-center"
      >
        <div
          className="PlusView__headingTitle"
        >
          hub_supporter_feature_more_title
        </div>
        <div
          className="PlusView__headingDescription"
        >
          hub_supporter_feature_more_description
        </div>
        <img
          alt="hub_supporter_feature_more_title"
          className="PlusView__featureImage more"
          src="/app/images/hub/plus/feature-more.svg"
        />
      </div>
    </div>
    <div
      className="PlusView--addPaddingTop"
    >
      <div
        className="row align-center-middle"
      >
        <div
          className="PlusView__bar flex-child-grow"
        />
        <a
          className="PlusView__button button PlusView--addSideMargin"
          href="https://checkout.ghosterystage.com/plus?utm_source=gbe&utm_campaign=intro_hub_plus"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          get_ghostery_plus
        </a>
        <div
          className="PlusView__bar flex-child-grow"
        />
      </div>
    </div>
  </div>
</div>
`;
